event.c

```diff
@@ -152,106 +152,139 @@ static void on_motion_detected_command(struct context *cnt,
-                MOTION_LOG(ERR, TYPE_DB, NO_ERRNO, "%s: Connection to PostgreSQL database '%s' failed: %s",
-                           cnt->conf.database_dbname, PQerrorMessage(cnt->database_pg));
+        if (PQstatus(cnt->database_pg) == CONNECTION_BAD) {
 
-                // This function will close the connection to the server and attempt to reestablish a new connection to the same server,
-                // using all the same parameters previously used. This may be useful for error recovery if a working connection is lost
-                PQreset(cnt->database_pg);
+            MOTION_LOG(ERR, TYPE_DB, NO_ERRNO, "%s: Connection to PostgreSQL database '%s' failed: %s",
+                       cnt->conf.database_dbname, PQerrorMessage(cnt->database_pg));
 
-                if (PQstatus(cnt->database_pg) == CONNECTION_BAD) {
-                    MOTION_LOG(ERR, TYPE_DB, NO_ERRNO, "%s: Re-Connection to PostgreSQL database '%s' failed: %s",
-                               cnt->conf.database_dbname, PQerrorMessage(cnt->database_pg));
-                } else {
-                    MOTION_LOG(INF, TYPE_DB, NO_ERRNO, "%s: Re-Connection to PostgreSQL database '%s' Succeed",
-                               cnt->conf.database_dbname);
-                }
+	    // This function will close the connection to the server and attempt to reestablish a new connection to the same server,
+	    // using all the same parameters previously used. This may be useful for error recovery if a working connection is lost
+            PQreset(cnt->database_pg);
 
-            } else if (PQresultStatus(res) != PGRES_COMMAND_OK) {
-                MOTION_LOG(ERR, TYPE_DB, SHOW_ERRNO, "%s: PGSQL query [%s] failed", sqlquery);
-                PQclear(res);
+            if (PQstatus(cnt->database_pg) == CONNECTION_BAD) {
+                MOTION_LOG(ERR, TYPE_DB, NO_ERRNO, "%s: Re-Connection to PostgreSQL database '%s' failed: %s",
+                           cnt->conf.database_dbname, PQerrorMessage(cnt->database_pg));
+            } else {
+                MOTION_LOG(INF, TYPE_DB, NO_ERRNO, "%s: Re-Connection to PostgreSQL database '%s' Succeed",
+                           cnt->conf.database_dbname);
             }
+
+        } else if (PQresultStatus(res) != PGRES_COMMAND_OK) {
+            MOTION_LOG(ERR, TYPE_DB, SHOW_ERRNO, "%s: PGSQL query [%s] failed", sqlquery);
+            PQclear(res);
         }
+    }
 #endif /* HAVE_PGSQL */
 
 #ifdef HAVE_SQLITE3
-        if ((!strcmp(cnt->conf.database_type, "sqlite3")) && (cnt->conf.database_dbname)) {
-            int res;
-            char *errmsg = 0;
-            res = sqlite3_exec(cnt->database_sqlite3, sqlquery, NULL, 0, &errmsg);
-            if (res != SQLITE_OK ) {
-                MOTION_LOG(ERR, TYPE_DB, NO_ERRNO, "%s: SQLite error was %s",
-                           errmsg);
-                sqlite3_free(errmsg);
-            }
+    if ((!strcmp(cnt->conf.database_type, "sqlite3")) && (cnt->conf.sqlite3_db)) {
```

Comment:
This looks like it was applied against a older version of the motion code.  The sqlite3_db configuration variable was eliminated and replaced with the more generic database_dbname variable (See code above this line that is being replaced)